gtdynamics/factors/ConstBiasFactor.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <gtsam/nonlinear/NonlinearFactor.h>
+
+#include <boost/assign/list_of.hpp>
+#include <boost/serialization/base_object.hpp>
+
+namespace gtsam {
+
+/** A factor that substitute certain variables of a base factor with constraint
+ * manifold variables.
+ * It is used for constraint manifold optimization, since the variables
+ * connected with constraint factors will be replaced by constraint manifold
+ * variables.
+ * Note that the noise model is stored twice (both in base factor and the
+ * noisemodel of substitute factor. The noisemodel in the base factor will be
+ * ignored. */
+class ConstBiasFactor : public NoiseModelFactor {
```

Comment:
We might need a better name for this, since we have a `ConstantBias` class in GTSAM and it can be confusing.